Core/src/TrackFitting/GainMatrixUpdater.cpp

```diff
@@ -58,10 +58,11 @@ std::tuple<double, std::error_code> GainMatrixUpdater::visitMeasurement(
     ACTS_VERBOSE("Gain Matrix K:\n" << K);
 
     if (K.hasNaN()) {
+      // set to error abort execution
       error = (direction == Direction::Forward)
```

Comment:
this is actually the only reason we pull in the direction which seems a bit odd to me. maybe we should drop that and generalize the error